auth0-extensions/auth0-gitlab-deploy
# Apiv4

# ✏️ Changes
The Auth Gitlab Extension was using Gitlab API v3. As of June 4 2018, access to Gitlab API v3 was disabled, which is why we just started seeing these issues. Updated the extension to use a more recent version of the `gitlab` module, which supports Gitlab API v4.

https://about.gitlab.com/2018/06/01/api-v3-removal-impending/

# 🔗 References
Zendesk: https://auth0.zendesk.com/agent/tickets/43083
CMS: https://auth0team.atlassian.net/browse/CMS-389

# 🎯 Testing
- Open already configured Gitlab extension
- Click Deploy button
- Error will no longer be thrown.

✅ This has been tested locally
✅ This has been tested in a webtask
✅This change has unit test coverage
✅This change has been acceptance tested
✅This change has been manually smoke tested

# 🚀 Deployment
🚫 This change can support multiple releases of the code serving traffic at the same time
✅ This can be deployed any time because it has no prerequisites.

# 🎡 Rollout

✅ The change will be available to the complete environment at once
🚫 The change is behind a feature flag and will go through a percentage based rollout process
🚫 The change is behind a flag and customers will opt-in to the change

In order to verify that the deployment was successful we will update the extensions via a test tenant and perform smoke tests to ensure proper execution.

# 🔥 Rollback
We will rollback if:

- the smoke test fails
- we receive a customer ticket indicating we broke the extension

## 📄 Procedure

A code revert+redeploy can be done to revert the change.

# 🖥 Appliance

✅ This change is compatible with the appliance.
